scripts/sst_prep_obs.sh

```diff
@@ -1,95 +1,132 @@
 #!/bin/bash -l 
 set -e
 
-while getopts "i:d:" opt; do
+while getopts "i:" opt; do
    case $opt in
-      i) sst_source=("$OPTARG");;
-      d) filename=("$OPTARG");;
+      i) SSTsource=("$OPTARG");;
    esac
 done
 shift $((OPTIND -1))
 
-echo SST SOURCE = ${sst_source}
+sat=`echo $SSTsource | awk -F 'sst.|_l3u*' '{print $2}'`
+datasource=`echo $SSTsource | awk -F '_' '{print $2}'`
 
 cd $DCOM_ROOT
 
-ObsRunDir=$RUNCDATE/Data/${CDATE}    #TODO: Should not be needed here ...
-OUTFILE=ioda.${sst_source}.${DA_SLOT_LEN}h.nc   #Filename of the processed obs
-PREPROCobs=${IODA_ROOT}/${CDATE}/${OUTFILE}     #FullPath/Filename of preprocessed obs
-PROCobs=${ObsRunDir}/${OUTFILE}                 #FullPath/Filename of observations to be ingested
+OUTFILE=ioda.sst.${sat}_${datasource}.${DA_SLOT_LEN}h.nc  #Filename of the processed obs
+PREPROCobs=${IODA_ROOT}/${CDATE}/${OUTFILE}               #FullPath/Filename of preprocessed obs
+PROCobs=${ObsRunDir}/${OUTFILE}                           #FullPath/Filename of observations to be ingested
 
-echo "obsrundir: "${ObsRunDir}
 echo "obsrundir: "$RUNCDATE/Data/${CDATE}
 
-#Check if the observations have been preprocessed.
+echo Processing $SSTsource
+case $SSTsource in 
+   "sst.windsat_l3u.ghrsst")
+      sat=WSAT
+      subsample=true
+      skip=5
+      ;; 
+   "sst.gmi_l3u.ghrsst")
+      sat=GMI
+      ;;
+   "sst.amsr2_l3u.ghrsst")
+      sat=AMSR2
+      ;;
+   "sst.viirs_l3u.nesdis")
+      sat=VIIRS              #TODO: Needs to be checked
+      ;;
+   "sst.avhrr19_l3u.nesdis")
+      sat=AVHRR19
+      subsample=true
+      skip=250
+      ;;
+   "sst.avhrrmta_l3u.nesdis")
+      sat=AVHRRMTA
+      subsample=true
+      skip=250
+      ;;
+esac
+
+#Check if the observations have been preprocessed. 
 if [ -f "${PREPROCobs}" ]; then
    echo
-   echo PreProcessed Observations are copied from "${PREPROCobs}" \
+   echo PreProcessed Observations for ${SSTsource} are copied from "${PREPROCobs}" \
         to ${PROCobs}
    echo
 
-   cp -rf ${PREPROCobs} ${PROCobs}
+   cp -rp ${PREPROCobs} ${PROCobs}
```

Comment:
what purpose does the -p serve in this case?

Reply:
preserves the date...

Reply:
kept the original version